src/components/App/App.jsx

```diff
@@ -182,16 +192,15 @@ class App extends React.Component {
   }
 
   editFilter(key) {
-    const filter = new Filter(key)
-    this.setState({ filter }, () => {
+    this.setState({ filter: key }, () => {
       ipcRenderer.send('title', `Edit Filter ${key}`)
       this.changeView('edit-filter')
     })
   }
 
   changeView(view) {
     window.scrollTo(0, 0)
-    this.setState({ view })
+    this.setState({ view, previousView: this.state.view })
```

Comment:
It's not! Not until after `this.setState` on this lines run will `this.state.view` be changed.